src/App.tsx

```diff
@@ -14,11 +15,108 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type Props = {};
+type State = {
+  isGoodsVisible: boolean,
+  goods: string[],
+  isReversed: boolean,
+  isSortedAlphabet: boolean,
+  visibleGoods: string[],
+  isSortedLength: boolean,
+};
+
+class App extends React.Component<Props, State> {
+  state: State = {
+    isGoodsVisible: false,
+    goods: [...goodsFromServer],
```

Comment:
You already have `visibleGoods` in your state, this property is unnecessary here